filters/terminal_punct.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+import sys
+from sys import stdin, stdout, stderr
+import argparse
+import math
+
+def parse_user_args():
+    """Parse the arguments necessary for this filter"""
+    parser = argparse.ArgumentParser(description="Filter segments based on penalty score with respecct to co-occurrence of terminal punctuation masrks ")
+    parser.add_argument("--threshold", default=-2, type=float)
+    parser.add_argument("--debug", action='store_true')
+    return parser.parse_args()
+
+
+def filter_terminal_punct(threshold: float, debug: bool=True) -> None:
+    """Filter segments based on penalty score with respecct to co-occurrence of terminal punctuation masrks """
+    
+    for line in stdin:
+        fields = line.strip().split('\t')
+        src = fields[-2].strip()
+        trg = fields[-1].strip()
+        
+        src_punct = len([c for c in src if c in ['.', '?', '!', '…']])
+        trg_punct = len([c for c in trg if c in ['.', '?', '!', '…']])
```

Comment:
```py
src_punct = len([c for c in src if c in {'.', '?', '!', '…'}])
```
Just replacing it with a set seems to be a bit faster. Faster than `sum(1…)` and moving the punctuation to a variable surprisingly…